### PR TITLE
Neo4j: make addAllDocumentsToNeo4j()

### PR DESCRIPTION
**Pull request by lindajiawenli**
#1156 

### Goal:

Create a function addAllDocumentsToNeo4j() that makes the appropriate nodes and edges in Neo4j for *ALL the documents in the "real" factoid database.

*with appropriate filters so that only public documents are used. Do not put demos, drafts, etc. into Neo4j.

### CODE REVIEW

**Comment by lindajiawenli** [comment]
Current approach: using the `getDocuments` function found in `src/server/routes/api/document/index.js`, which seems appropriate for what I need to do (I want to get an array of document instances and call each of them with the previously made/tested `addDocumentToNeo4j`)

Running into a bug: 
`Error: Expected at least one of 'cache', 'factory', 'factoryOptions' to be defined`

**Comment by lindajiawenli** [comment]
After running `node -r esm src/neo4j/add-all-docs.js` and then opening `http://localhost:7474/browser/` (this is Neo4j Browser, from a docker container), run `MATCH (n) RETURN n` in the command line and this is what you will see:

Screen Shot 2023-03-31 at 2 41 34 PM

File: src/neo4j/neo4j-document.js [hunk 1]
```diff
@@ -166,3 +169,32 @@ export async function addDocumentToNeo4j(doc) {
 
   return;
 }
+
+export async function addAllDocumentsToNeo4j() {
+  let rdbConn;
+  let testDb;
```
Comment (jvwong) [inline src/neo4j/neo4j-document.js:175]: Here's where we need to have a configurable database instance that could access  non-test (production) data.

 In practice, probably need to hook into those used by a running factoid server instance: `db` in `src/server/db.js` which uses environment variables in `src/config.js` which could be overwritten at runtime.

**Review by jvwong — CHANGES_REQUESTED** [review]

**Comment by jvwong** [comment]
One caveat:  if you use any existing functions that touch the database (e.g. `get Documents` in `src/server/routes/api/document/index.js`) it uses the database settings defined on the factoid server 

https://github.com/PathwayCommons/factoid/blob/906ce07f089578f35c20e6a627e61bc70ee9bea2/src/config.js#L51-L57

**Comment by jvwong** [comment]
I'm starting to reconsider the necessity of this at all: If `addAllDocumentsToNeo4j` is just a for loop around `addDocumentToNeo4j`, what benefit does it provide with respect to the use case(s)?

File: src/neo4j/neo4j-document.js [hunk 2]
```diff
@@ -166,3 +169,32 @@ export async function addDocumentToNeo4j(doc) {
...
+  const dbName = 'factoid';
+  const dbTables = ['document', 'element'];
+  const opts = { limit: null, offset: 0 };
+
+  try {
+    const { total, results } = await getDocuments(opts);
```
[the remark this answers is not in the document]
Comment (lindajiawenli) [reply src/neo4j/neo4j-document.js:181]: Yeah, I did notice that, and I would've liked to get Document instances directly, but I wasn't quite sure how to do that and `getDocuments` seemed too convenient for me to not use.

Is there a function that does that and I simply missed it, or would I have to make one?

**Comment by jvwong** [comment]
Let's close and I'll follow up on #1157- probably reuse this code there.

**Pull request closed without merging**